### ISSUE TITLE
Uninitialized default execution space in serial unit test

### ISSUE THREAD

**Issue by dsunder** [kokkos/kokkos#579]
When building the serial unit tests with both Cuda and Serial enabled, the default execution space is Cuda which is never initialized.

Short term solution: Don't use default execution spaces in device specific unit tests.
Longer term solution: Only enable the necessary spaces in device specific unit tests.

**Comment by crtrott** [comment]
I don't believe that this can be the case since we actually test that combination. And if I do: 
```
KOKKOS_PATH/generate_makefile.bash --with-serial --with-cuda
make test
```
everything works including cuda and serial. 
What did you do to get an issue?

**Comment by dsunder** [comment]
This error occurred in an MDRange unit test that used the default execution space.  The short term fix was to stop using the default space.

**Comment by crtrott** [comment]
So are you saying the MDRange tests are not run as part of the normal testing?

**Comment by ndellingwood** [comment]
The MDRange tests are run as part of normal testing. 

I ran into this with the unit tests I was working on for atomic views while testing with Cuda and Serial enabled. When compiling with a single space there were no compiler errors, only if enabling for example Cuda and Serial. 

Initially in some functors I had used `parallel_for` while passing only the size for the range rather than a range policy, e.g. `parallel_for( N , functor);` 
When building the tests, Cuda tests built fine, but then the Serial tests failed with the following error when passing the range size to the parallel_for:
`Kokkos::Cuda::Cuda instance constructor : ERROR device not initialized`

Replacing the range size with a range policy templated on the space fixed the problem since the `parallel_for` had access to the `RangePolicy` template arguments. The mistake I had initially made in the functors (passing range size rather than a range policy templated on the appropriate space) demonstrates this issue though.

**Comment by crtrott** [comment]
Yeah but this is not an issue in the development branch. All those tests pass ...

**Comment by ndellingwood** [comment]
I fixed the tests (atomic views) before issuing the pull request (else it would fail the tests with test_all_sandia script). I added the info above to describe a specific encounter with this issue. If there is reason to investigate further, the error described can be replicated by replacing the `RangePolicy` with the range `length` in a `parallel_for` in any of the atomic view test functors, e.g. line 427 of core/unit_tests/TestAtomicViews.hpp and building with cuda and serial enabled.
